python/lsst/sims/utils/htmModule.py

```diff
@@ -1152,3 +1268,78 @@ def halfSpaceFromRaDec(ra, dec, radius):
     dd = np.cos(np.radians(radius))
     xyz = cartesianFromSpherical(np.radians(ra), np.radians(dec))
     return HalfSpace(xyz, dd)
+
+
+def halfSpaceFromPoints(pt1, pt2, pt3):
+    """
+    Return a Half Space defined by two points on a Great Circle
+    and a third point contained in the Half Space.
+
+    Parameters
+    ----------
+    pt1, pt2 -- two tuples containing (RA, Dec) in degrees of
+    the points on the Great Circle defining the Half Space
+
+    pt3 -- a tuple containing (RA, Dec) in degrees of a point
+    contained in the Half Space
+
+    Returns
+    --------
+    A Half Space
+    """
+
+    vv1 = cartesianFromSpherical(np.radians(pt1[0]), np.radians(pt1[1]))
+    vv2 = cartesianFromSpherical(np.radians(pt2[0]), np.radians(pt2[1]))
+    axis = np.array([vv1[1]*vv2[2]-vv1[2]*vv2[1],
+                     vv1[2]*vv2[0]-vv1[0]*vv2[2],
+                     vv1[0]*vv2[1]-vv1[1]*vv2[0]])
+
+    axis /= np.sqrt(np.sum(axis**2))
+
+    vv3 = cartesianFromSpherical(np.radians(pt3[0]), np.radians(pt3[1]))
+    if np.dot(axis, vv3)<0.0:
+        axis *= -1.0
+
+    return HalfSpace(axis, 0.0)
+
+def intersectHalfSpaces(hs1, hs2):
+    """
+    Parameters
+    ----------
+    hs1, hs2 are Half Spaces
+
+    Returns
+    -------
+    A list of the cartesian points where the Half Spaces intersect
+
+    Based on section 3.5 of
+    Szalay A. et al. (2007)
+    "Indexing the Sphere with the Hierarchical Triangular Mesh"
+    arXiv:cs/0701164
+    """
```

Comment:
Should we include a case where the half spaces are the same? Are we assuming the user won't do this? Or am I missing it in here?

Reply:
That case will be encompassed in the "Half spaces are based on parallel planes that don't intersect" case at line ~ 1320

I take your point, though, that returning an empty list might not be the right thing to do there.

Reply:
Would it be sufficient to just add a line to the docstring explaining that the list will be empty if the half spaces are identical?

Reply:
Yeah, that would be fine. Just a statement to help somebody understand what would happen in that case before using it is good.